core/src/main/scala/org/apache/spark/storage/BlockManagerMasterActor.scala

```diff
@@ -522,7 +523,9 @@ private[spark] class BlockManagerInfo(
         logInfo("Removed %s on %s on tachyon (size: %s)".format(
           blockId, blockManagerId.hostPort, Utils.bytesToString(blockStatus.tachyonSize)))
       }
+      return BlockStatus(storageLevel, 0, 0, 0)
     }
+    null
```

Comment:
actually, can you explain the `null` case?  how does that happen, and won't that result in an NPE when it gets to your code in `StorageStatusListener` and in `JsonProtocol`?